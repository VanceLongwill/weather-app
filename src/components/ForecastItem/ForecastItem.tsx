import * as React from 'react'

import { IForecast } from '../../types'

interface IForecastItemProps {
  data: IForecast
}

export function formatTemperature(temp: number) {
  const roundedTemp = Math.round(temp)
  return `${roundedTemp}Â°C`
}

export function ForecastItem(props: IForecastItemProps) {
  const { data } = props
  return (
    <>
      <td>{data.displayName}</td>
      {data.list.map((item, i) => {
        return (
          <td key={`${data.displayName}-${i}-${item.main.grnd_level}`}>
            {formatTemperature(item.main.temp)}
          </td>
        )
      })}
    </>
  )
}
