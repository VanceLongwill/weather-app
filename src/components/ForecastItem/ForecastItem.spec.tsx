import { shallow } from 'enzyme'
import * as React from 'react'

import { ForecastItem, formatTemperature } from './ForecastItem'

import stateSub from '../../__test__/stateStub'

describe('ForecastItem', () => {
  it('should match snapshot', () => {
    const component = shallow(
      <ForecastItem data={stateSub.forecast.locations[0]} />
    )
    expect(component).toMatchSnapshot()
  })

  it('should render the correct number of cells', () => {
    const stubLocation = stateSub.forecast.locations[0]
    const component = shallow(<ForecastItem data={stubLocation} />)
    expect(component.find('td')).toHaveLength(stubLocation.list.length + 1)
  })

  it('formatTemperature should format temperatures correctly', () => {
    const tests = [
      {
        input: 14.76,
        output: '15°C',
      },
      {
        input: 1.01,
        output: '1°C',
      },
    ]

    tests.forEach(test => {
      expect(formatTemperature(test.input)).toBe(test.output)
    })
  })
})
