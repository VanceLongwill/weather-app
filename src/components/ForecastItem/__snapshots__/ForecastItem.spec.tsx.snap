// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastItem should match snapshot 1`] = `
<Fragment>
  <td>
    London
  </td>
  <td
    key="London-0-1008.68"
  >
    10°C
  </td>
  <td
    key="London-1-1010.79"
  >
    8°C
  </td>
  <td
    key="London-2-1011.44"
  >
    7°C
  </td>
  <td
    key="London-3-1011.49"
  >
    7°C
  </td>
  <td
    key="London-4-1012.52"
  >
    7°C
  </td>
  <td
    key="London-5-1013.93"
  >
    10°C
  </td>
  <td
    key="London-6-1014.62"
  >
    12°C
  </td>
  <td
    key="London-7-1014.93"
  >
    13°C
  </td>
</Fragment>
`;
